src/main/java/com/fasterxml/jackson/databind/ser/std/MapSerializer.java

```diff
@@ -913,6 +933,41 @@ public void serializeFieldsUsing(Map<?,?> value, JsonGenerator gen, SerializerPr
         }
     }
 
+    public void serializeFieldsUsing(Map<?, ?> value, JsonGenerator gen, SerializerProvider provider,
+                                     JsonSerializer<Object> ser, Object bean)
+            throws IOException {
+        final JsonSerializer<Object> keySerializer = _keySerializer;
+        final TypeSerializer typeSer = _valueTypeSerializer;
+        Set<String> fields = Arrays.stream(bean.getClass().getDeclaredFields()).map(Field::getName).collect(Collectors.toSet());
```

Comment:
I am sorry but this is not going to work for most cases plus adds significant overhead for every usage.